# Notebook 1: Exploración y Selección de Variables IPM
## Monitor de Vulnerabilidad Económica - Colombia

**Objetivo:** Explorar el dataset consolidado GEIH 2024 y seleccionar las variables relacionadas con el cálculo del Índice de Pobreza Multidimensional (IPM).

**Dataset:** `data/processed/geih_2024_consolidated.csv`

---


In [1]:
# Importar librerías
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from pathlib import Path

# Configuración
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)
sns.set_style('darkgrid')
plt.rcParams['figure.figsize'] = (12, 6)

print("✓ Librerías importadas")


✓ Librerías importadas


In [4]:
# Cargar dataset consolidado
df = pd.read_csv('../data/processed/geih_2024_consolidated.csv', low_memory=False)

print(f"Dataset cargado:")
print(f"  Registros: {df.shape[0]:,}")
print(f"  Columnas: {df.shape[1]:,}")
print(f"  Memoria: {df.memory_usage(deep=True).sum() / 1024**2:.2f} MB")


Dataset cargado:
  Registros: 688,507
  Columnas: 575
  Memoria: 3650.00 MB


In [5]:
# Ver estructura básica
print("Primeras 3 filas:")
df.head(3)


Primeras 3 filas:


PERIODO  MES   PER  DIRECTORIO  SECUENCIA_P  ORDEN  HOGAR  REGIS  AREA  \
0  20240101    1  2024     7655976            1      1      1     10   5.0   
1  20240101    1  2024     7655976            1      2      1     10   5.0   
2  20240101    1  2024     7655976            1      3      1     10   5.0   

   CLASE      FEX_C18  DPTO  PT  P6016  P3271  P6040  P6030S1  P6030S3  P6050  \
0      1  2540.569858     5   1      1      1     39      5.0   1984.0      1   
1      1  2540.569858     5   1      2      2     32      2.0   1991.0      2   
2      1  2540.569858     5   1      3      2      3     11.0   2020.0      3   

   P6083  P6083S1  P6081  P6081S1  P2057  P2059  P2061  P6080 P6080S1  \
0      2      NaN      2      NaN      2    2.0      2      6     NaN   
1      2      NaN      2      NaN      2    2.0      2      6     NaN   
2      1      2.0      2      NaN      2    2.0      2      6     NaN   

   P6080S1A1  P6070  P6071  P6071S1  P3147S1  P3147S2  P3147S3  P3147S4  \
0        NaN    2.0    1.0      2.0      NaN      NaN      NaN      NaN   
1        NaN    2.0    1.0      1.0      NaN      NaN      NaN      NaN   
2        NaN    NaN    NaN      NaN      NaN      NaN      NaN      NaN   

   P3147S5  P3147S6  P3147S7  P3147S8  P3147S9  P3147S10  P3147S11 P3147S10A1  \
0      NaN      NaN      NaN      NaN      NaN       NaN       1.0        NaN   
1      NaN      NaN      NaN      NaN      NaN       NaN       1.0        NaN   
2      NaN      NaN      NaN      NaN      NaN       NaN       NaN        NaN   

   P6090  P6100  P6110    P6120  P1906S1  P1906S2  P1906S3  P1906S4  P1906S5  \
0      1    1.0    9.0      NaN        4        4        4        4        4   
1      1    1.0    1.0  46500.0        4        4        4        4        4   
2      1    1.0    5.0      NaN        4        4        4        4        4   

   P1906S6  P1906S7  P1906S8  P6160  P6170  P3041  P3042  P3042S1  P3042S2  \
0        4        4        4    1.0    2.0    NaN    5.0      1.0      NaN   
1        4        4        4    1.0    2.0    NaN    5.0      2.0      NaN   
2        4        4        4    2.0    2.0    NaN    1.0      0.0      NaN   

   P3043  P3043S1  P3038  P3039  POB_MAY18  MES_hogar  PER_hogar  REGIS_hogar  \
0    1.0      NaN    2.0    1.0        1.0          1       2024            1   
1    2.0      NaN    1.0    2.0        1.0          1       2024            1   
2    NaN      NaN    NaN    NaN        NaN          1       2024            1   

   AREA_hogar  CLASE_hogar  FEX_C18_hogar  DPTO_hogar  P4000  P4010  P4020  \
0         5.0            1    2540.569858           5      2      1      4   
1         5.0            1    2540.569858           5      2      1      4   
2         5.0            1    2540.569858           5      2      1      4   

   P4030S1  P4030S1A1  P4030S2  P4030S3  P4030S4  P4030S4A1  P4030S5  P70  \
0        1        1.0        1        1        1        2.0        1    1   
1        1        1.0        1        1        1        2.0        1    1   
2        1        1.0        1        1        1        2.0        1    1   

   P5000  P5010  P5020  P5030  P5040  P5050  P5070  P5080  P5090 P5090S1  \
0      2      1      1    1.0      1      1      1    3.0      3     NaN   
1      2      1      1    1.0      1      1      1    3.0      3     NaN   
2      2      1      1    1.0      1      1      1    3.0      3     NaN   

   P5100  P5110  P5130     P5140  P5222S1  P5222S2  P5222S3  P5222S4  P5222S5  \
0    NaN    NaN    NaN  550000.0      NaN      1.0      NaN      NaN      NaN   
1    NaN    NaN    NaN  550000.0      NaN      1.0      NaN      NaN      NaN   
2    NaN    NaN    NaN  550000.0      NaN      1.0      NaN      NaN      NaN   

   P5222S6  P5222S7  P5222S8  P5222S9  P5222S10 P5222S8A1  P6008  \
0      NaN      NaN      NaN      NaN       NaN       NaN      3   
1      NaN      NaN      NaN      NaN       NaN       NaN      3   
2      NaN      NaN      NaN    

## Variables del IPM según metodología DANE

El IPM se calcula con 5 dimensiones y 15 indicadores:

### 1. **EDUCACIÓN Y CAPITAL HUMANO** (4 indicadores)
- Bajo logro educativo
- Analfabetismo
- Inasistencia escolar
- Rezago escolar

### 2. **CONDICIONES DE LA NIÑEZ Y LA JUVENTUD** (4 indicadores)
- Trabajo infantil
- No acceso a cuidado de la primera infancia
- Barreras a servicios para cuidado de la primera infancia
- Barreras de acceso a servicio de salud

### 3. **TRABAJO** (2 indicadores)
- Desempleo de larga duración
- Empleo informal

### 4. **SALUD** (2 indicadores)
- Sin aseguramiento en salud
- Barreras de acceso a servicio de salud

### 5. **ACCESO A SERVICIOS PÚBLICOS DOMICILIARIOS Y CONDICIONES DE LA VIVIENDA** (3 indicadores)
- Sin acceso a fuente de agua mejorada
- Inadecuada eliminación de excretas
- Material inadecuado de pisos, paredes o techos

---


In [6]:
# Identificar variables disponibles en el dataset
# Listar todas las columnas
print(f"Total columnas: {len(df.columns)}\n")
print("Columnas disponibles:")
print(list(df.columns))


Total columnas: 575

Columnas disponibles:
['PERIODO', 'MES', 'PER', 'DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR', 'REGIS', 'AREA', 'CLASE', 'FEX_C18', 'DPTO', 'PT', 'P6016', 'P3271', 'P6040', 'P6030S1', 'P6030S3', 'P6050', 'P6083', 'P6083S1', 'P6081', 'P6081S1', 'P2057', 'P2059', 'P2061', 'P6080', 'P6080S1', 'P6080S1A1', 'P6070', 'P6071', 'P6071S1', 'P3147S1', 'P3147S2', 'P3147S3', 'P3147S4', 'P3147S5', 'P3147S6', 'P3147S7', 'P3147S8', 'P3147S9', 'P3147S10', 'P3147S11', 'P3147S10A1', 'P6090', 'P6100', 'P6110', 'P6120', 'P1906S1', 'P1906S2', 'P1906S3', 'P1906S4', 'P1906S5', 'P1906S6', 'P1906S7', 'P1906S8', 'P6160', 'P6170', 'P3041', 'P3042', 'P3042S1', 'P3042S2', 'P3043', 'P3043S1', 'P3038', 'P3039', 'POB_MAY18', 'MES_hogar', 'PER_hogar', 'REGIS_hogar', 'AREA_hogar', 'CLASE_hogar', 'FEX_C18_hogar', 'DPTO_hogar', 'P4000', 'P4010', 'P4020', 'P4030S1', 'P4030S1A1', 'P4030S2', 'P4030S3', 'P4030S4', 'P4030S4A1', 'P4030S5', 'P70', 'P5000', 'P5010', 'P5020', 'P5030', 'P5040', 'P5050', 'P5070

In [7]:
# Definir variables clave del IPM según documentación DANE
# Basado en el diccionario de datos GEIH

ipm_variables = {
    # IDENTIFICADORES
    'identificadores': ['PERIODO', 'DIRECTORIO', 'SECUENCIA_P', 'ORDEN', 'HOGAR'],
    
    # DEMOGRÁFICAS BÁSICAS
    'demograficas': ['P6040', 'P6016', 'P6050', 'P6080', 'AREA', 'CLASE', 'DPTO'],
    
    # EDUCACIÓN (Dimensión 1)
    'educacion': ['P6160', 'P6170', 'P6210', 'P6220'],
    
    # SALUD (Dimensión 2 y parte de Dimensión 4)
    'salud': ['P6090', 'P6100', 'P6110', 'P6120'],
    
    # TRABAJO (Dimensión 3)
    'trabajo': ['FT', 'P6240', 'P6250', 'P6430', 'P6920'],
    
    # VIVIENDA (Dimensión 5)
    'vivienda': ['P5000', 'P5010', 'P5020', 'P5030', 'P5040'],
    
    # SERVICIOS PÚBLICOS (Dimensión 5)
    'servicios': ['P5070', 'P5080', 'P5090', 'P5100', 'P5110'],
    
    # NIÑEZ Y JUVENTUD (Dimensión 4)
    'ninez': ['P6170S1', 'P6210S1', 'P6585S1', 'P6585S2', 'P6585S3']
}

# Combinar todas las variables
all_ipm_vars = []
for category, vars_list in ipm_variables.items():
    all_ipm_vars.extend(vars_list)

print(f"Variables IPM identificadas: {len(all_ipm_vars)}")
print(f"\nPor categoría:")
for category, vars_list in ipm_variables.items():
    print(f"  {category}: {len(vars_list)} variables")


Variables IPM identificadas: 40

Por categoría:
  identificadores: 5 variables
  demograficas: 7 variables
  educacion: 4 variables
  salud: 4 variables
  trabajo: 5 variables
  vivienda: 5 variables
  servicios: 5 variables
  ninez: 5 variables


In [8]:
# Verificar qué variables IPM están disponibles en el dataset
available_ipm_vars = [var for var in all_ipm_vars if var in df.columns]
missing_ipm_vars = [var for var in all_ipm_vars if var not in df.columns]

print(f"Variables IPM disponibles: {len(available_ipm_vars)}/{len(all_ipm_vars)}")
print(f"\nVariables FALTANTES: {len(missing_ipm_vars)}")
if missing_ipm_vars:
    print("Faltantes:", missing_ipm_vars)


Variables IPM disponibles: 36/40

Variables FALTANTES: 4
Faltantes: ['P6210', 'P6220', 'P6170S1', 'P6210S1']


In [9]:
# Agregar variables de ingresos para calcular vulnerabilidad económica
# Estas son esenciales para definir el target

income_variables = ['INGLABO', 'INGTOT', 'INGTOTOB', 'INGTOTES']

# Verificar disponibilidad
available_income = [var for var in income_variables if var in df.columns]
print(f"Variables de ingresos disponibles: {len(available_income)}/{len(income_variables)}")
print("Disponibles:", available_income)

# Agregar a la lista de variables a extraer
all_selected_vars = list(set(available_ipm_vars + available_income))
print(f"\n✓ Total variables seleccionadas: {len(all_selected_vars)}")


Variables de ingresos disponibles: 1/4
Disponibles: ['INGLABO']

✓ Total variables seleccionadas: 37


In [10]:
# Crear dataset con variables IPM seleccionadas
df_ipm = df[all_selected_vars].copy()

print(f"Dataset IPM creado:")
print(f"  Registros: {df_ipm.shape[0]:,}")
print(f"  Variables: {df_ipm.shape[1]}")
print(f"  Memoria: {df_ipm.memory_usage(deep=True).sum() / 1024**2:.2f} MB")


Dataset IPM creado:
  Registros: 688,507
  Variables: 37
  Memoria: 194.36 MB


In [11]:
# Análisis de valores nulos por variable
print("Análisis de valores nulos:\n")
null_analysis = pd.DataFrame({
    'Variable': df_ipm.columns,
    'Nulos': df_ipm.isnull().sum(),
    '% Nulos': (df_ipm.isnull().sum() / len(df_ipm) * 100).round(2)
}).sort_values('% Nulos', ascending=False)

print(null_analysis[null_analysis['% Nulos'] > 0].head(15))


Análisis de valores nulos:

        Variable   Nulos  % Nulos
P5100      P5100  663816    96.41
P6585S3  P6585S3  532604    77.36
P6585S1  P6585S1  532604    77.36
P6585S2  P6585S2  532604    77.36
P6120      P6120  516476    75.01
P5110      P5110  446329    64.83
P6250      P6250  417145    60.59
INGLABO  INGLABO  406793    59.08
P6430      P6430  394214    57.26
P6920      P6920  394214    57.26
P6110      P6110  387721    56.31
FT            FT  355395    51.62
AREA        AREA  192278    27.93
P6240      P6240  142463    20.69
P6100      P6100   26573     3.86


In [12]:
# Ver primeras filas del dataset IPM
print("Primeras 5 filas del dataset IPM:\n")
df_ipm.head()


Primeras 5 filas del dataset IPM:



P6040  P5110  P6110  P5020  P6585S2  P6016    P6120  P6585S3  P6080  P6050  \
0     39    NaN    9.0      1      NaN      1      NaN      NaN      6      1   
1     32    NaN    1.0      1      1.0      2  46500.0      1.0      6      2   
2      3    NaN    5.0      1      NaN      3      NaN      NaN      6      3   
3     39    NaN    1.0      1      1.0      1  46500.0      2.0      6      1   
4     22    NaN    NaN      1      1.0      2      NaN      2.0      6      3   

   DIRECTORIO  P6090  P6430    INGLABO   PERIODO   FT  P6920  P5100  P5000  \
0     7655976      1    NaN        NaN  20240101  NaN    NaN    NaN      2   
1     7655976      1    1.0  1160000.0  20240101  1.0    1.0    NaN      2   
2     7655976      1    NaN        NaN  20240101  NaN    NaN    NaN      2   
3     7655977      1    1.0  1600000.0  20240101  1.0    1.0    NaN      3   
4     7655977      2    1.0  1500000.0  20240101  1.0    2.0    NaN      3   

   DPTO  P5080  P5040  HOGAR  CLASE  AREA  P5090  P6250  P6240  P5030  \
0     5    3.0      1      1      1   5.0      3    2.0    4.0    1.0   
1     5    3.0      1      1      1   5.0      3    NaN    1.0    1.0   
2     5    3.0      1      1      1   5.0      3    NaN    NaN    1.0   
3     5    3.0      1      1      1   5.0      3    NaN    1.0    1.0   
4     5    3.0      1      1      1   5.0      3    NaN    1.0    1.0   

   P6585S1  P6160  P6100  ORDEN  P6170  P5070  P5010  SECUENCIA_P  
0      NaN    1.0    1.0      1    2.0      1      1            1  
1      2.0    1.0    1.0      2    2.0      1      1            1  
2      NaN    2.0    1.0      3    2.0      1      1            1  
3      2.0    1.0    1.0      1    2.0      1      1            1  
4      2.0    1.0    NaN      2    2.0      1      1            1

In [14]:
# Guardar dataset IPM
output_file = '../data/processed/geih_2024_ipm_variables.csv'
df_ipm.to_csv(output_file, index=False)

print(f"✓ Dataset IPM guardado en: {output_file}")
print(f"  Tamaño del archivo: {Path(output_file).stat().st_size / 1024**2:.2f} MB")


✓ Dataset IPM guardado en: ../data/processed/geih_2024_ipm_variables.csv
  Tamaño del archivo: 70.83 MB


## Resumen de Exploración

✅ **Completado:**

1. **Dataset cargado**: 688,507 registros, 575 columnas
2. **Variables IPM identificadas**: ~40 variables relacionadas con las 5 dimensiones del IPM
3. **Variables de ingresos**: Identificadas para calcular vulnerabilidad económica
4. **Dataset IPM creado**: Solo con variables relevantes
5. **Análisis de nulos**: Identificados para tratamiento posterior

**Próximo paso:** 
- Notebook 2: Feature Engineering y Preparación de Datos
  - Limpieza de datos
  - Tratamiento de nulos
  - Creación de target (vulnerabilidad económica)
  - Split train/test

---
